File: URI/uri-1961/python-uri-1961/jumping_frog.py
first_line = input().split(" ")
jump_height, number_of_pipes = [int(i) for i in first_line]

heights_line = input().split()
heights_line = [int(i) for i in heights_line]

win = True

last_pipe_height = heights_line[0]
for next_pipe_height in heights_line:
    if((last_pipe_height - next_pipe_height > jump_height) or
       (next_pipe_height - last_pipe_height > jump_height)):

        win = False

    last_pipe_height = next_pipe_height

if(win):
    print("YOU WIN")
else:
    print("GAME OVER")

File: URI/uri-1961/cpp-uri-1961/jumping_frog.cpp
#include <iostream>
 
using namespace std;
 
int main() {
 
	int jump_height, number_of_pipes;
	cin >> jump_height >> number_of_pipes;

	bool win = true;
	int last_pipe_height, next_pipe_height;

	for(int i = 0; i < number_of_pipes; i++){
		cin >> next_pipe_height;
		if(i == 0) 
			last_pipe_height = next_pipe_height;
		
		if((last_pipe_height - next_pipe_height > jump_height) ||
		   (next_pipe_height - last_pipe_height > jump_height))

			win = false;
		
		last_pipe_height = next_pipe_height;
	}
	if(win)
		cout << "YOU WIN\n";
	else
		cout << "GAME OVER\n";
 
	return 0;
}
